### DISCUSSION TITLE
Pass the entire response instead of the statusCode to window.livewire.onError

### DISCUSSION THREAD

**Discussion by lucasromanojf** [livewire/livewire#3730] [3. Feature ideas]
Hey!

I think that it would be better to pass the entire response to the window.livewire.onError callback instead of the statusCode only. 

The reason is that sometimes a different message is returned from backend for a same status code (e.g.: a JSON response with an error message that varies). So we can't show the correct message since we only have the statusCode.

**Comment by joshhanley** [comment]
@lucasromanojf yep that makes sense, PR's are always welcome 🙂

**Reply by lucasromanojf** [reply]
Hi @joshhanley, cool, gonna do a PR next week.

**Reply by lucasromanojf** [reply]
PR #4179

**Comment by lucasromanojf** [comment]
PR #4179 merged

**Reply by andcl** [reply]
Hello, just one question ahead. I am trying to use this functionality to customize the error handling of my webapp through Livewire events. Currently, I have:
```
Livewire.onError(function (status, response) {
    Livewire.emit('livewireException', response);
    return false;  // To avoid default modal behaviour
});
```
But `console.log(response)` just prints a basic JS `Response` object from where I can get the `response.status` (HTTP error code, e.g. 404) and `response.statusText` (e.g. standard 'Not found' text (for the 404 example)). I am obvoiusly interested in the specific message returned from the backend to customize the alert I am going to show, but I can't find it anywhere in the so-called JS `Response` object. 
Any ideas? 

Thanks in advance.

**Reply by silnarm** [reply]
@andcl @victorybiz 
You can read the response body, which is what I believe you want, by using the json() or text() (as appropriate) methods on the response.
These methods return promises though, so your code should be structured like so:
```
Livewire.onError((status, response) => {
    response.text().then((body) => {
        console.log(body);
    });
    return false;
});
```

**Comment by victorybiz** [comment]
@andcl Did you find a solution or workaround to getting messages returned from the backend?

**Reply by njxqlus** [reply]
Did you?
